### DISCUSSION TITLE
[Bug]: Issue with spatie/laravel-pdf on Live Server (Linux)

### DISCUSSION THREAD

**Discussion by Sirloko** [spatie/laravel-pdf#75] [Q&A]
### What happened?

I've installed the spatie/laravel-pdf package in my application, and it works fine locally. However, when I deploy it to a live server (Linux) and install all the required dependencies, attempting to generate a PDF results in an attached file of zero bytes.

[test.pdf](https://github.com/spatie/laravel-pdf/files/14164366/test.pdf)

### How to reproduce the bug

use function Spatie\LaravelPdf\Support\pdf;

return pdf()
        ->html('<p>Hello world!!</p>')
        ->name('test.pdf')
        ->download('test.pdf');

### Package Version

1.1

### PHP Version

8.2

### Laravel Version

10.3.2

### Which operating systems does with happen with?

Linux

**Comment by valkyriweb** [comment]
I am having the same issue, not sure where I am going wrong here. Nothing in the log file

**Comment by aldinokemal** [comment]
How many RAM and processors are in the server?
probably the cause of that.

**Reply by Sirloko** [reply]
I am running it on a  4GB LAMP Droplet

**Comment by valkyriweb** [comment]
I resolved this issue by updating the composer and node packages; the path to the puppeteer executable wasn't correct it was using a different version path to the one that was installed on my server.

**Reply by Sirloko** [reply]
How can I check the path puppeteer is using? I already updated the node and composer packages

**Reply by valkyriweb** [reply]
Hey Sirloko,

I found it by looking in the .cache directory on forge for puppeteer, it had a reference to the installed chrome path, my chrome version was different on my local versus on the server.

Once I got that to match up it worked.

Kind Regards,
Luke

**Comment by josecaseiro** [comment]
I'm still getting this error.
Does anyone found solution?

**Reply by dennykurniawan48** [reply]
Hi, see my [solution](https://github.com/spatie/laravel-pdf/discussions/150#discussioncomment-9785365).
